scripts/prune-extra-keep-files.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+for EXERCISE_DIRECTORY in $(find exercises -mindepth 2 -maxdepth 2 -type d -name "src"); do
+  STARTER_DIRECTORY="${EXERCISE_DIRECTORY}/main/java"
+  STARTER_FILE_COUNT=$(find "${STARTER_DIRECTORY}" -mindepth 1 -maxdepth 1 -type f -name "*.java" | wc -l)
```

Comment:
👍 appreciate the precision you're using, here; it's easy for declarative statements to "say more" than we intend.